src/components/textarea/bl-textarea.ts

```diff
@@ -0,0 +1,277 @@
+   */
+  @property({ type: Boolean, attribute: 'label-fixed' })
+  labelFixed = false;
+
+  /**
+   * Sets placeholder of the textarea
+   */
+  @property({})
+  placeholder?: string;
+
+  /**
+   * Enables showing character counter.
+   */
+  @property({ type: Boolean, attribute: 'character-counter' })
+  characterCounter = false;
+
+  /**
+   * Adds help text
+   */
+  @property({ type: String, attribute: 'help-text' })
+  helpText?: string;
+
+  /**
+   * Set custom error message
+   */
+  @property({ type: String, attribute: 'invalid-text' })
+  customInvalidText?: string;
+
+  /**
+   * Sets minimum length of the textarea
+   */
+  @property({ type: Number })
+  minlength?: number;
+
+  /**
+   * Sets max length of textarea
+   */
+  @property({ type: Number })
+  maxlength?:number;
+
+  /**
+   * Sets initial value of the textarea
+   */
+  @property()
+  value = '';
+
+  /**
+   * Sets textarea visible row count.
+   */
+  @property({ type: Number})
+  rows?: number = 4;
+
+  @event('bl-input') private onInput: EventDispatcher<string>;
+
+  @event('bl-change') private onChange: EventDispatcher<string>;
+
+  @event('bl-invalid') private onInvalid: EventDispatcher<ValidityState>;
+
+
+  private initialHeight:number;
+  private lineHeight:number;
+  private expandScroll = false;
+
+  constructor() {
+    super();
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener('invalid', this.onError);
+
+
+    this.internals.form?.addEventListener('submit', () => {
+      this.reportValidity();
+    });
+  }
+
+
+  private onError = (): void => {
+    this.onInvalid(this.internals.validity);
+  }
+
+  private  inputHandler(event: Event) {
+    if(this.expand){
+      this.autoResize();
+    }
+
+
+    const value = (event.target as HTMLTextAreaElement).value;
+
+    this.setValue(value);
+    this.onInput(value);
+  }
+
+  private changeHandler(event: Event) {
+    const value = (event.target as HTMLTextAreaElement).value;
+
+    this.dirty = true;
+    this.setValue(value);
+    this.onChange(value);
+  }
+
+  firstUpdated() {
+    this.setValue(this.value);
+  }
+
+  updateSlotted(event:Event) {
+    const _target = (event.target as HTMLSlotElement)
+    const value = _target.assignedNodes().map((n) => n.textContent).join('');
+    this.value = value
+    this.setValue(this.value);
+  }
+
+  reportValidity() {
+    this.dirty = true;
+    return this.checkValidity();
+  }
+
+  valueChangedCallback(value: string): void {
+    this.value = value;
+  }
+
+  validityCallback(): string | void {
+    return this.customInvalidText || this.validationTarget?.validationMessage;
+  }
+
+  private autoResize() {
+    const verticalPadding = parseInt(getComputedStyle(this.validationTarget).padding[0]) * 2;
+    const scrollHeight = this.validationTarget.scrollHeight;
+    if(!this.initialHeight) {
+      this.initialHeight = scrollHeight;
+      this.lineHeight = (this.initialHeight-verticalPadding) / this.rows!;
```

Comment:
If we really need line height, I think we can read it from a variable, instead of trying to calculate it (we set line-height with related font definition in our design system)

Reply:
Is using fixed value for line height not problem for different browser or similar cases?

Reply:
I think we really need this because of computing current line